Lib/tkinter/__init__.py

```diff
@@ -300,6 +300,31 @@ def _get_default_root(what=None):
     return _default_root
 
 
+def _get_temp_root():
+    global _support_default_root
+    if not _support_default_root:
+        raise RuntimeError("No master specified and tkinter is "
+                           "configured to not support default root")
+    root = _default_root
+    if root is None:
+        assert _support_default_root
+        _support_default_root = False
+        root = Tk()
+        _support_default_root = True
+        assert _default_root is None
+        root.withdraw()
+        root._temporary = True
```

Comment:
Do you actually *want* to raise an exception instead of just making a temporary non-default root?
```suggestion
    if not _support_default_root:
        root = Tk()
        root.withdraw()
        root._temporary = True
    else:
        root = _default_root
        if root is None:
            _support_default_root = False
            root = Tk()
            _support_default_root = True
            assert _default_root is None
            root.withdraw()
            root._temporary = True
```
By saving _support_default_root, the logic could be rearranged to remove duplication, but I would redo the tests first.

Reply:
Yes, I do. If the default root window is disabled, you should always specify master/parent/root. Otherwise you will get an error. It is a programming error.